core/src/main/java/org/apache/iceberg/PositionDeletesTable.java

```diff
@@ -196,15 +265,14 @@ protected List<String> scanColumns() {
      */
     public BatchScan baseTableFilter(Expression expr) {
       return new PositionDeletesBatchScan(
-          table(), schema(), context(), Expressions.and(baseTableFilter, expr));
+          table(), schema(), context(), Expressions.and(baseTableFilter, expr), fieldMap);
     }
 
     @Override
     protected CloseableIterable<ScanTask> doPlanFiles() {
       String schemaString = SchemaParser.toJson(tableSchema());
-
-      // prepare transformed partition specs and caches
-      Map<Integer, PartitionSpec> transformedSpecs = transformSpecs(tableSchema(), table().specs());
+      Map<Integer, PartitionSpec> transformedSpecs =
```

Comment:
We use the transformed spec to evaluate the filters, so we need to bind against the new (reassigned) field ids,  hence passing the field id map here.